src/UI/Components/ErrorView/ErrorView.tsx

```diff
@@ -25,27 +28,54 @@ export const ErrorView: React.FC<Props> = ({
   retry,
   delay,
   ...props
-}) => (
-  <Delayed delay={delay}>
-    <EmptyState {...props}>
-      <EmptyStateHeader
-        titleText={<>{title || words("error")}</>}
-        icon={<EmptyStateIcon icon={ExclamationTriangleIcon} />}
-        headingLevel="h4"
-      />
-      <EmptyStateBody>
-        <StyledErrorMessage>{message}</StyledErrorMessage>
-      </EmptyStateBody>
-      <EmptyStateFooter>
-        {retry && (
-          <Button variant="primary" onClick={retry}>
-            {words("retry")}
-          </Button>
-        )}
-      </EmptyStateFooter>
-    </EmptyState>
-  </Delayed>
-);
+}) => {
+  const { authController } = useContext(DependencyContext);
```

Comment:
Why here? And not in the baseApiHelper? Now you are adding an additional dependency into this file, making it high cohesion instead of the opposite. 

I understand better why you have a glitch now.
The useEffect will run after the initial mount of the component, and then update accordingly. This takes a fragment of a second, but enough time to actually display the error view instead of redirecting. 

The redirection should be done as high level as possible, right when the response lands into the code.